### ISSUE TITLE
Multidimensionality concepts domains

### ISSUE THREAD

**Issue by vit1251** [behave/behave#428]
There are many possible Given, When and Then, and it would make sense to separate them on semantic group (by requirement document, version or another criteria) in the directory tree.

Let's I show you filesystem directory tree as exmaple:

```
 +
 +-- features
 |      +-- steps (with steps impl scope "global")
 |      +-- 01_user_authority.feature
 |      +-- 02_user_guest_main_page_abilities.feature
 |      +-- payment_operation
 |                  +-- steps (with steps impl scope "global" + "payment_operation")
 |                  +-- 03_user_payment_make_order.feature
```

this make implementation in separate scope "global" and "payment_operation" ... and have exception when try to use impl from level up scope ... it make possible to sure that no where then this scope no more use this implementation and does not trashing global scope of domain.

What you think about that?

**Comment by jenisys** [comment]
First, I am not quite sure I understood why you think you need this directory structure. 

Second: In general, it is a good idea provide a hierarchical structure to counter to much complexity in one place. But this does not work with the current implementation (yet). You basically need an external or own runner implementation to use it for such a structure. I am working on a improved runner that provides a "scope concept", but it is not usable yet. The "scope concept" provides own scopes (or namespaces) for the ``steps`` directory or an ``environment.py`` file and you automatically inherit the parent scopes. Basically steps and type converters (and hooks) are registered in own namespaces.

**Comment by vit1251** [comment]
> I am not quite sure I understood why you think you need this directory structure.

To separate different implementation scope i.e. "When I press button" may implemented as press button on "Android device" or "Android device emulator" it different implementation but similar scenarios.

P.S. Another benefits is run this scope by different parameters for example let's imagine in global scope:

```
  Given A few device 
      | Name    | Directory with implementation |
      | Android | android_impl                            |
      | iOS        | iOS_impl                                 |
      | WP9      | wp9_impl                                 |
   When I start scope "payment_operation"
   Then I have no error in this block
```

**Comment by jenisys** [comment]
Mmh, that "Scenario" is executed only once, but you probably mean the right thing (Scenario Outline Examples table).

One thing that you can already use for that is the "test stage" concept (option: ``--stage``; see "New and Noteworthy" section of the documentation). But using it, would mean multiple test runs, one for each stage. Test stages make sense when you have different step implementations (and "environment.py"), normally one for each team with the same features (or a subset of these features). In addition, such a concept was suggested also for testing at different layers of a system (for example: view/GUI layer and model layer testing). The feature description is independent of the layer interface. It just states what is intended to do, but not how it is done (which is implemented in the steps normally).

**Comment by aisbaa** [comment]
Hi, I also need to specify multiple steps directories. Would be nice to have something like `behave --steps /path/to/steps/ ./features/`, this would load steps from `/path/to/steps/` and `./features/steps/`. @jenisys could you share your progress on improved runner that provides a "scope concept"?

Also it seems that current runner is able to load steps from multiple directories, please see [runner.py:719](https://github.com/behave/behave/blob/master/behave/runner.py#L719). The only problem is that its impossible to pass extra_step_paths from command line.
